src/App.tsx

```diff
@@ -16,47 +16,93 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [
+    sortedGoods,
+    setSortedGoods,
+  ] = useState(goodsFromServer);
+  const [isVisibleReser, setIsVisibleReser] = useState(false);
+
+  enum SortType {
+    Alphabet = 'alphabet',
+    Length = 'length',
+    Reverse = 'reverse',
+  }
+
+  function handleSort(filter: string) {
+    let sorted: string[] = [];
+
+    if (filter === SortType.Alphabet) {
+      sorted = [...sortedGoods].sort();
+    }
+
+    if (filter === SortType.Length) {
+      sorted = [...sortedGoods].sort((a, b) => a.length - b.length);
+    }
+
+    if (filter === SortType.Reverse) {
+      sorted = [...sortedGoods].reverse();
+    }
```

Comment:
You make a copy of the array in three cases, so it is worth doing it once when creating the array at the beginning of this function.